### ISSUE TITLE
How to get http status and detailed response from an API, whether the API returns an error or not?

### ISSUE THREAD

**Issue by mregnauld** [aws-amplify/amplify-ios#1065]
In my AWS project, I created webservices using Lambda and API Gateway. Those webservices are called by my iOS app, using the Amplify framework.

In my lambda functions:

- I return something like the following when there is no error:

```Javascript
	return {
		statusCode: 200,
		body: JSON.stringify({
			"example_key_1": "example_value",
			"example_key_2": 123456789
		})
	};
```

- I return something like the following when there is an error:

```Javascript
	return {
		statusCode: 400,
		body: JSON.stringify({
			"custom_error_code": 333333
		})
	};
```

And from my iOS app, I call my API by doing something like the following:

```Swift
	Amplify.API.post(request: request) { result in
		switch result
		{
			case .success(let data):
				print("success: \(data)")
			case .failure(let error):
				print("error: \(error)"
		}
	}
```

Now, here are the infos I need to get from the response of the API:
- the status code, whether there is an error or not
- the body of the response, whether there is an error or not

In other words, I need to get in my iOS app the whole content of each `return` of the lambda, when I call the API.

It seems easily doable with the Amplify framework for Javascript, according to [this][1], but I can't find the equivalent for iOS.

How can I achieve that with the Amplify framework for iOS?

  [1]: https://github.com/aws-amplify/amplify-js/issues/4144

**Comment by royjit** [comment]
Closing this as duplicate - https://github.com/aws-amplify/amplify-ios/issues/655, please follow the issue 655 for updates.
